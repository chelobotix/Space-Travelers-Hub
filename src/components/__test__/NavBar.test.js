import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../Header/Navbar';

describe('NavBar', () => {
  test('Check if render', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    const navbar = screen.getByRole('navigation');
    expect(navbar).toBeInTheDocument();
  });

  test('NavBar Tree', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
