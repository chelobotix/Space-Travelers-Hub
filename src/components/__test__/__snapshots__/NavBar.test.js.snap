// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar Tree 1`] = `
<nav
  className="nav"
>
  <ul
    className="ul"
  >
    <li>
      <a
        className="logo"
        href="/rockets"
        onClick={[Function]}
      >
        <img
          alt=""
          className="img"
          src="planet.png"
        />
        <p
          className="logoName"
        >
          Space Travelers' Hub
        </p>
      </a>
    </li>
    <li
      className="right"
    >
      <a
        className="normalLink"
        href="/rockets"
        onClick={[Function]}
        style={Object {}}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="normalLink"
        href="/missions"
        onClick={[Function]}
        style={Object {}}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="normalLink border"
        href="/profile"
        onClick={[Function]}
        style={Object {}}
      >
        Profile
      </a>
    </li>
  </ul>
</nav>
`;
